run-tests.pl

```diff
@@ -128,10 +141,22 @@ sub usage
 {
    my ( $exitcode ) = @_;
 
-   print STDERR <<'EOF';
+   my @output_formats =
+      map { $_->FORMAT }
+      grep { $_->can( "FORMAT" ) } output_formats();
+
+   my @homeserver_implementations =
+      map { $_->name() } homeserver_factories();
```

Comment:
Perhaps rather than a `sub name` that generates it from the class name, each HS implementation class could be given a

    use constant NAME => "Synapse";

if the HS impl class names don't directly correspond.